### ISSUE TITLE
Add more information for ModuleManager and OwnerManager

### ISSUE THREAD

**Issue by rmeissner** [gnosis/safe-docs#3]
It would be nice to have separate section for the ModuleManager and OwnerManager contract.

For the ModuleManager it should be outlined how modules are stored and managed. Also shortly explain one module to show how the interaction with the safe works and what needs to be implemented (e.g. replay protection)

For the OwnerManager it should be outlined how owners are stored and managed. Also mention scenarios where transactions might fail.

**Comment by rmeissner** [comment]
Example of failing scenario by @apanizo 👍 
```
1. Create a Safe with owners: Foo, Bar, and threshold 2
          mapping owners -->
                  (0x1, Foo)
                  (Foo, Bar)
                  (Bar, 0x1)
2. Generate a pending transaction for removing owner Foo
3. Generate and confirm transaction (both owners) of adding owner Moe
          mapping owners -->
                  (0x1, Moe)
                  (Moe, Foo)
                  (Foo, Bar)
                  (Bar, 0x1)
4. User Bar goes to confirm (and therefore execute) tx generated in STEP 2.... IMPOSSIBLE TO EXECUTE
```

**Comment by rmeissner** [comment]
Closed in favor of #39
